lib/travis/logs/services/process_log_part.rb

```diff
@@ -107,6 +108,11 @@ def run
         end
         alias find_or_create_log log_id
 
+        private def normalize_number
+          return unless payload['number'] == 'last'
+          payload['number'] = database.max_log_part_number_for_log(log_id)
```

Comment:
Agreed.  I'm not sure what to do about the racy situation, and I'm pretty sure it already exists in production.  The use of `last` is meant to replace cases where other applications write a "final message" log part, such as when a job is auto-cancelled when a new PR build arrives.  What do you think about using `database.max_log_part_number_for_log(log_id) + 1000` (or some other reasonably big number)?